build/storage/tests/it/ptf_tests/tests/test_hot_plug.py

```diff
@@ -0,0 +1,149 @@
+        )
+        self.storage_target_platform.create_subsystem()
+        self.assertFalse(
+            self.storage_target_platform.is_port_free(self.tests_config.nvme_port)
+        )
+        self.assertTrue(
+            self.storage_target_platform.is_app_listening_on_port(
+                "spdk_tgt", self.tests_config.nvme_port
+            )
+        )
+
+        volumes_ids = self.storage_target_platform.create_ramdrives(
+            self.tests_config.min_ramdrive
+        )
+        self.assertEqual(len(volumes_ids), self.tests_config.min_ramdrive)
+
+        self.assertEqual(
+            self.host_target_platform.get_number_of_virtio_blk_devices(), 0
+        )
+        devices_handles = self.ipu_storage_platform.create_virtio_blk_devices(
+            volumes_ids
+        )
+        self.assertEqual(
+            self.host_target_platform.get_number_of_virtio_blk_devices(),
+            self.tests_config.min_ramdrive,
+        )
+
+        for device_handle in devices_handles:
+            self.ipu_storage_platform.delete_virtio_blk_device(device_handle)
+        self.assertEqual(
+            self.host_target_platform.get_number_of_virtio_blk_devices(), 0
+        )
+
+    def tearDown(self):
+        self.ipu_storage_platform.clean()
+        self.storage_target_platform.clean()
+        self.host_target_platform.clean()
+
+
+class TestMaxHotPlug(BaseTest):
+    def setUp(self):
+        platforms_factory = KVMPlatformsFactory()
+        self.storage_target_platform = platforms_factory.create_storage_target_platform()
+        self.ipu_storage_platform = platforms_factory.create_ipu_storage_platform()
+        self.host_target_platform = platforms_factory.create_host_target_platform()
+        self.tests_config = TestsConfig()
+
+    def runTest(self):
+        self.assertTrue(
+            self.storage_target_platform.is_port_free(self.tests_config.nvme_port)
+        )
+        self.storage_target_platform.create_subsystem()
+        self.assertFalse(
+            self.storage_target_platform.is_port_free(self.tests_config.nvme_port)
+        )
+        self.assertTrue(
+            self.storage_target_platform.is_app_listening_on_port(
+                "spdk_tgt", self.tests_config.nvme_port
+            )
+        )
+
+        volumes_ids = self.storage_target_platform.create_ramdrives(
+            self.tests_config.max_ramdrive
+        )
+        self.assertEqual(len(volumes_ids), self.tests_config.max_ramdrive)
+
+        self.assertEqual(
+            self.host_target_platform.get_number_of_virtio_blk_devices(), 0
+        )
+        devices_handles = self.ipu_storage_platform.create_virtio_blk_devices(
+            volumes_ids
+        )
+        self.assertEqual(
+            self.host_target_platform.get_number_of_virtio_blk_devices(),
+            self.tests_config.max_ramdrive,
+        )
+
+        for device_handle in devices_handles:
```

Comment:
this for loop has one disadvantage. It won't cleanup the resources if the assert above throws an exception.
I saw that you placed TODO in clean method. But I 'd place one more TODO here, that this for loop can be moved into ipu_platform clean method and ipu_platform will track all allocated devices to perform that cleanup.

Reply:
changed